src/lib.rs

```diff
@@ -994,7 +994,7 @@ where
 
     fn prepare_request<RE>(self) -> Result<HttpRequest, RequestTokenError<RE, TE>>
     where
-        RE: Fail,
+        RE: Error,
```

Comment:
Not sure if we have to add `Send + Sync + 'static` bounds here. `Fail` trait had them, but tests seem to work fine without them.